rfcs/card-component.md

```diff
@@ -0,0 +1,56 @@
+# Card Component
+
+## Exports
+
+* `Card`
+* `StyledAction`
+* `StyledBody`
+* `StyledContents`
+* `StyledHeaderImage`
+* `StyledRoot`
+* `StyledThumbnail`
+* `StyledTitle`
+
+## `Card` API
+
+* `action: React.Node` - Optional
+  Contents of the `StyledAction` component
+* `children: React.Node` - Optional
+  Contents of the `StyledBody` component
+* `hasThumbnail: ({+thumbnail?: string}) => boolean` - Optional
+  Function that takes `Card` props and returns a boolean that represents if a thumbnail will be rendered
+* `headerImage: string` - Optional
+  `src` of the `StyledHeaderImage` component
+* `overrides: {Action, Body, Contents, HeaderImage, Root, Thumbnail, Title}` - Optional
+  * `Action: OverrideObjectT` - Optional
+  * `Body: OverrideObjectT` - Optional
+  * `Contents: OverrideObjectT` - Optional
+  * `HeaderImage: OverrideObjectT` - Optional
+  * `Root: OverrideObjectT` - Optional
+  * `Thumbnail: OverrideObjectT` - Optional
+  * `Title: OverrideObjectT` - Optional
+    Overrides for presentational components.
+* `thumbnail: string` - Optional
+  `src` of the `StyledThumbnail` component
+* `title: React.Node` - Optional
+  Contents of the `StyledTitle` component
+
+## Usage
+
+Basic usage:
+
+```javascript
+import * as React from 'react';
+import {Card} from 'baseui/card';
+
+export default () => <Card title="This is a card" />;
```

Comment:
Might be good to flesh this out a little bit since we're currently using our RFCs as documentation for the time being. We've also started moving "Usage" to the top for that reason (#195).

Reply:
Definitely- I'll flesh this out a bit more today. I mostly wanted to get the functional component up for review before doing so since the props/composable API is a bit of a departure from other things we've done. I'll add some more here and some tests today.